application/src/main/java/net/furizon/backend/feature/badge/controller/BadgeController.java

```diff
@@ -0,0 +1,47 @@
+package net.furizon.backend.feature.badge.controller;
+
+import jakarta.validation.constraints.NotNull;
+import lombok.RequiredArgsConstructor;
+import net.furizon.backend.feature.badge.dto.BadgeUploadResponse;
+import net.furizon.backend.feature.badge.usecase.DeleteUserBadgeUsecase;
+import net.furizon.backend.feature.badge.usecase.UploadUserBadgeUsecase;
+import net.furizon.backend.infrastructure.security.FurizonUser;
+import net.furizon.backend.infrastructure.usecase.UseCaseExecutor;
+import org.springframework.http.MediaType;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.util.Set;
+
+@RestController
+@RequestMapping("/api/v1/badge")
+@RequiredArgsConstructor
+public class BadgeController {
+    private final UseCaseExecutor useCaseExecutor;
+
+    @PostMapping(value = "/upload", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    public BadgeUploadResponse upload(
```

Comment:
We better generalize the scope of the upload. There won't be only single user uploading, we already need it for fursuit badges, too.

In my client it would sound better if there was a central media upload endpoint, since the upload is enclosed in the scope of an upload component

then a save button will receive the media id that will be set in the correct place